wait_requests

```diff
@@ -14,14 +14,25 @@ output_file=$1
 
 echo "Monitoring CloudWatch metrics for new datapoints..."
 
+# Check if localstack-main container is running
+if [ -z "$(docker ps -q -f name=localstack-main)" ]; then
+    echo "Error: LocalStack container is not running."
+    exit 1
+fi
+
+# Get the start time of the localstack container
+start_time=$(docker inspect -f '{{.State.StartedAt}}' localstack-main | sed 's/\.[0-9]*Z//' | xargs -I{} date -u -j -f '%Y-%m-%dT%H:%M:%S' {} +'%Y-%m-%dT%H:%M:%S')
+end_time=$(date -u +'%Y-%m-%dT%H:%M:%S')
+
 # Initial run to get the current number of datapoints
-current_datapoints=$(aws cloudwatch get-metric-statistics --namespace ServerlessDataProcessingPipeline/Latencies --metric-name Latency --start-time 2022-01-01T00:00:00 --end-time 2025-01-02T00:00:00 --period 1 --statistics SampleCount | jq '[.Datapoints[].SampleCount] | add')
+current_datapoints=$(aws cloudwatch get-metric-statistics --namespace ServerlessDataProcessingPipeline/Latencies --metric-name Latency --start-time $start_time --end-time $end_time --period 1 --statistics SampleCount | jq '[.Datapoints[].SampleCount] | add')
```

Comment:
is there a reason for using `period 1` when in the end you sum all `SampleCount` anyhow?
If you would choose e.g. `period 60` it would aggregate all datapoints in the given timeframe to 1-minute intervals and return the SampleCount. This way you also don't hit the limit of 1440 datapoints that fast.

Reply:
@steffyP just wanting to get the granularity on a per-second basis to see the actual throughput per sec. As long as this is run within the first 24 minutes after having started LS, this shouldn't be a problem. Again, this is just for a simple benchmark script.